src/block/block_cache.c

```diff
@@ -428,8 +426,8 @@ __wt_blkcache_put(WT_SESSION_IMPL *session, wt_off_t offset, size_t size, uint32
      * If more than the configured fraction of the file is likely to fit into the OS buffer cache,
      * don't use this cache.
      */
-    if (blkcache->system_ram >=
-      __blkcache_estimate_filesize(session) * blkcache->fraction_in_os_cache) {
+    if (__blkcache_estimate_filesize(session) <
+      (blkcache->system_ram * 100) / blkcache->percent_file_in_os_cache) {
         WT_STAT_CONN_INCR(session, block_cache_bypass_put);
```

Comment:
For clarity, I would change this to: 

```
 if ((__blkcache_estimate_filesize(session) * blkcache->percent_file_in_os_cache) / 100 <
       blkcache->system_ram * 100))
```

This way the formula matches to how we explain this in the comment.